src/Microsoft.OData.Core/TypeUtils.cs

```diff
@@ -103,6 +103,11 @@ internal static void ParseQualifiedTypeName(string qualifiedTypeName, out string
             }
 
             int separator = qualifiedTypeName.LastIndexOf(".", StringComparison.Ordinal);
+            if (separator == -1)

```

Comment:
I assume it's for reading. 
How about to writing? It's writing the full-qualified type name in the context Url?